src/source_sampling.cpp

```diff
@@ -211,6 +187,25 @@ void pyne::Sampler::setup() {
   }
   else throw std::invalid_argument("Mesh file must contain only tets or hexes.");
 
+  if (ve_type == moab::MBHEX){
+      // cell_number_tag
+      if (tag_names.find("cell_number_tag_name") == tag_names.end()) {
```

Comment:
Can this be done with a `try` block that throws an error when the value isn’t found in the dictionary?  Or does the new mesh infrastructure just create this if it’s not found?

Reply:
For Cartesian mesh, there is a `cell_number` and `cell_fracs` tags. However, for unstructured mesh, there isn't. Therefore, this block of code should in `if` but not `try` block.

Reply:
I think the top level if block stays, but then try within that

Reply:
Oops - this is not python... leave it how it is...